src/ng/httpBackend.js

```diff
@@ -46,7 +46,7 @@ function createHttpBackend($browser, createXhr, $browserDefer, callbacks, rawDoc
     url = url || $browser.url();
 
     if (lowercase(method) == 'jsonp') {
-      var callbackId = '_' + (callbacks.counter++).toString(36);
+      var callbackId = '_' + Math.random().toString(36).substring(7);
```

Comment:
According to the "birthday paradox", if you have 100,000 callbacks, you will have a 6% chance of a collision.  (See https://instacalc.com/28845, with "number of days" = 78364164096 = 36^7, and "number of people" = 100000.)  So, I think this needs to be a longer random number.

Reply:
Since we only care for the callback id while the request is in flight, we care about the number of concurrent JSONP requests. If you had 100000 of them, then 6% callback id collision rate would be the least of your problems 😛